hw/ip/spi_device/data/spi_device_testplan.hjson

```diff
@@ -146,5 +165,143 @@
       milestone: V2
       tests: []
     }
+    {
+      name: tpm_fifo_read
+      desc: '''
+            - Set tpm_mode to 0 in the TPM configuration register.
+            - Set en to 1 in the TPM configuration register.
+            - Randomise other fields in the TPM configuration register.
+
+            TODO: Clarify purpose of TPM_ACCESS_x registers in the spec.
+
+            - Send TPM read command over the SPI bus with a randomised address.
+              [TPM READ 1]
+            - Check TPM_STATUS.cmdaddr_notempty, INTR_STATE.tpm_header_notempty
+              and tpm_header_not_empty interrupt - they should be asserted if
+              hw_reg_dis == 0. [TPM READ 2,3]
+
+            TODO: Clarify in spec "The SW prepares the register value and
+                  writes the value into the read FIFO." [TPM READ 4]
+
+            - Confirm that the TPM submodule sends WAIT until the read FIFO is
+              available.
+            - When available, confirm that the TPM submodule sends START
+              followed by the register value.
+            - Compare this value with the expected value.
+
+            TODO: Clarify tpm_mode FIFO mode within the register table.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_fifo_write
+      desc: '''
+            - Set tpm_mode to 0 in the TPM configuration register.
+            - Set en to 1 in the TPM configuration register.
+            - Randomise other fields in the TPM configuration register.
+
+            TODO: Clarify purpose of TPM_ACCESS_x registers in the spec.
+
+            - Send TPM write command with a randomised address.
+
+            TODO: Clarify the difference between FIFO and CRB mode, can the
+                  command not be written into CMD_ADDR?
```

Comment:
@eunchan IIUC, for write, FIFO and CRB mode are the same. SPI host can't write directly to the register. the write data is always stored in the write fifo for SW to read, right?

Reply:
If so, in the latest amendment I have removed one of the write tests.